pkg/sandbox/infra.go

```diff
@@ -68,3 +85,100 @@ func PauseCommand(cfg *libconfig.Config, image *v1.Image) ([]string, error) {
 	cmd = append(cmd, image.Config.Cmd...)
 	return cmd, nil
 }
+
+func (s *sandbox) createResolvPath(podContainer *storage.ContainerInfo) (retErr error) {
+	// set DNS options
+	if s.config.DNSConfig == nil {
+		return nil
+	}
+
+	dnsServers := s.config.DNSConfig.Servers
+	dnsSearches := s.config.DNSConfig.Searches
+	dnsOptions := s.config.DNSConfig.Options
+	s.resolvPath = fmt.Sprintf("%s/resolv.conf", podContainer.RunDir)
+	err := ParseDNSOptions(dnsServers, dnsSearches, dnsOptions, s.resolvPath)
+	defer func() {
+		if retErr != nil {
+			if err := os.RemoveAll(s.resolvPath); err != nil {
```

Comment:
This should be `os.Remove` (as no recursion is intended) + ignoring `os.IsNotExist` error (as os.Remove can return it).